Extensions/XMPPMUCSUB/XMPPMUCSUB.h

```diff
@@ -0,0 +1,264 @@
+
+/**
+ * Implement this method when calling [mucsubInstanse failedToSubscribeToRoomJID:withError:]. It will be invoked if
+ * the request for subscribing roomJID is unsuccessfully executed or receives an unsuccessful response.
+ *
+ * @param sender XMPPMUCSUB object invoking this delegate method.
+ * @param roomJID room JID that user wants to subscribe to.
+ * @param error NSError containing more details of the failure.
+ */
+- (void)xmppMUCSUB:(XMPPMUCSUB *)sender failedToSubscribeToRoomJID:(XMPPJID *)roomJID withError:(NSError *)error;
+
+/**
+ * Implement this method when calling [mucsubInstanse didUnsubscribeFromRoomJID:]. It will be invoked if
+ * the request for unsubscribing from roomJID is successfully executed and receives a successful response.
+ *
+ * @param sender XMPPMUCSUB object invoking this delegate method.
+ * @param roomJID room JID that user wants to unsubscribe from.
+ */
+- (void)xmppMUCSUB:(XMPPMUCSUB *)sender didUnsubscribeFromRoomJID:(XMPPJID *)roomJID;
+
+/**
+ * Implement this method when calling [mucsubInstanse failedToUnsubscribeFromRoomJID:withError:]. It will be invoked if
+ * the request for unsubscribing from roomJID is unsuccessfully executed or receives an unsuccessful response.
+ *
+ * @param sender XMPPMUCSUB object invoking this delegate method.
+ * @param roomJID room JID that user wants to unsubscribe from.
+ * @param error NSError containing more details of the failure.
+ */
+- (void)xmppMUCSUB:(XMPPMUCSUB *)sender failedToUnsubscribeFromRoomJID:(XMPPJID *)roomJID withError:(NSError *)error;
+
+/**
+ * Implement this method when calling [mucsubInstanse didFetchSubscriptionList:]. It will be invoked if
+ * the request for fetching subscription list is successfully executed and receives a successful response.
+ *
+ * @param sender XMPPMUCSUB object invoking this delegate method.
+ * @param subscriptions list of subscriptions.
+ */
+- (void)xmppMUCSUB:(XMPPMUCSUB *)sender didFetchSubscriptionList:(NSXMLElement *)subscriptions;
+
+/**
+ * Implement this method when calling [mucsubInstanse failedToFetchSubscriptionListWithError:]. It will be invoked if
+ * the request for fetching subscription list is unsuccessfully executed or receives an unsuccessful response.
+ *
+ * @param sender XMPPMUCSUB object invoking this delegate method.
+ * @param error NSError containing more details of the failure.
+ */
+- (void)xmppMUCSUB:(XMPPMUCSUB *)sender failedToFetchSubscriptionListWithError:(NSError *)error;
+
+/**
+ * Implement this method when calling [mucsubInstanse didFetchSubscribersList:forRoomJID:]. It will be invoked if
+ * the request for fetching subscribers list is successfully executed and receives a successful response.
+ *
+ * @param sender XMPPMUCSUB object invoking this delegate method.
+ * @param roomJID room JID that user wants to fetch subscribers list for.
+ */
+- (void)xmppMUCSUB:(XMPPMUCSUB *)sender didFetchSubscribersList:(NSXMLElement *)subscriptions forRoomJID:(XMPPJID *)roomJID;
```

Comment:
Match to either SubscriptionList or SubscribersList

Reply:
SubscriptionList is list of everything you are subscribed to. SubscribersList is list of all users that are subscribed to a room.

Reply:
The comment wasn't correct.. I corrected it.